File: csharp/DesiArchitect.DesignPatterns/07_Facade/After/PnrService.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Facade.After;

public class PnrService : IPnrService
{
    public string Generate(BookingRequest request)
    {
        var pnr = $"PNR-{Random.Shared.Next(100000, 999999)}";
        Console.WriteLine($"  🎫 PnrService: Generated {pnr} for {request.PassengerName}");
        return pnr;
    }
}

File: typescript/src/07-facade/after/pnr-service.ts
import type { BookingRequest } from '../../shared/models.js';
import type { IPnrService } from './i-pnr-service.js';

export class PnrService implements IPnrService {
  generate(request: BookingRequest): string {
    const pnr = `PNR-${Math.floor(100000 + Math.random() * 900000)}`;
    console.log(`  🎫 PnrService: Generated ${pnr} for ${request.passengerName}`);
    return pnr;
  }
}